runners/direct-java/src/test/java/org/apache/beam/runners/direct/InProcessPipelineRunnerParDoTests.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.direct;
+
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.ListCoder;
+import org.apache.beam.sdk.coders.VarIntCoder;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.util.IllegalMutationException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Tests of {@link ParDo} when executed in the {@link InProcessPipelineRunner}.
+ */
+public class InProcessPipelineRunnerParDoTests {
```

Comment:
I'm concerned for a somewhat bureaucratic reason: since this is not of the form _className_`Test` it is a bit less discoverable. I understand that this convention is actually (deliberately) not a mechanical rule, but it is so common as to nearly be that way. Add to that the existence of `ParDoInProcessEvaluatorTest` and the waters are quite muddied. Could these be within the purview of `InProcessPipelineRunnerTest`?

If I were to be pedantic about this being a "unit" test, I'd be asking what unit is under test here. More pragmatically, I'm not sure what code I'd jump to if these tests were to fail.